### PR TITLE
feat(agents): add devils advocate agent

### PR DESCRIPTION
**Pull request by anschnapp**
## Pull Request Checklist

- [x] I have read and followed the [CONTRIBUTING.md](https://github.com/github/awesome-copilot/blob/main/CONTRIBUTING.md) guidelines.
- [x] My contribution adds a new instruction, prompt, or chat mode file in the correct directory.
- [x] The file follows the required naming convention.
- [x] The content is clearly structured and follows the example format.
- [x] I have tested my instructions, prompt, or chat mode with GitHub Copilot.
- [x] I have run `npm start` and verified that `README.md` is up to date.

---

## Description

Add devils-advocate agent for stress-testing ideas

This agent challenges user concepts by playing devil's advocate, finding flaws and edge cases. 

The workflow includes focused challenging of ideas one objection at a time, followed by an "end game" summary evaluating how well the idea held up, and finally transitions to an expert discussion mode where the agent objectively discusses the topic as a senior developer. 

The tone is intentionally direct and a bit combative to make discussions engaging, but remains constructive without being mean.

---

## Type of Contribution

- [ ] New instruction file.
- [ ] New prompt file.
- [x] New chat mode file.
- [ ] New collection file.
- [ ] Update to existing instruction, prompt, chat mode, or collection.
- [ ] Other (please specify):

---

## Additional Notes

None

---

By submitting this pull request, I confirm that my contribution abides by the [Code of Conduct](../CODE_OF_CONDUCT.md) and will be licensed under the MIT License.

### CODE REVIEW

File: agents/devils-advocate.agent.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+description: "I play the devil's advocate to challenge and stress-test your ideas by finding flaws, risks, and edge cases"
+tools: ['read', 'search', 'web']
+---
```
[the remark this answers is not in the document]
Comment (anschnapp) [reply agents/devils-advocate.agent.md:4]: the agent is not optimized for any model, so i guess it's correct to leave the model out

File: agents/devils-advocate.agent.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+description: "I play the devil's advocate to challenge and stress-test your ideas by finding flaws, risks, and edge cases"
```
[the remark this answers is not in the document]
Comment (anschnapp) [reply agents/devils-advocate.agent.md:2]: the suggested changes have issues with quoting and also a double apostrophe. so for me the source version looks correct.

File: agents/devils-advocate.agent.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+**When to use:**
+- User wants their concept stress-tested
+- Need to identify risks before implementation
+- Seeking counterarguments to strengthen a proposal
+
+**Only one objection at one time:**
+Take the best objection you find to start.
+Come up with a new one if the user is not convinced by it.
+
+**Challenging tone**:
```
Comment (anschnapp) [inline agents/devils-advocate.agent.md]: on a second thought i'm not sure about the challenging tone anymore.

personally i find it makes more fun to argue against the LLM here if the tone is challenging.
but this might not be something for everyone.

not that someone really feels offended by this...

is there a way to somehow make this optional. maybe comment it out here.
otherwise it might be the safer to leave it out here (sure i have to retest the agent then a bit)

Comment (codemillmatt) [reply agents/devils-advocate.agent.md]: I don't think this is that bad. If you want you could maybe reword it to something like "challenge assumptions. make sure we think through non-obvious scenarios. Have an honest and curious conversation with me - but don't be rude".

I don't know - your call here.

**Review by codemillmatt — COMMENTED** [review]
FYI on question on the "challenging" part of the agent

File: agents/devils-advocate.agent.md (hunk 3 above)
Comment (anschnapp) [reply agents/devils-advocate.agent.md]: I reworded a bit but took your suggestion.

After new testing the finish summary and expert mode discussion didn't work that good anymore (for whatever reasons). I have described them now in more detail and it worked better as before.

also i put now a little introduction into the agent. which tells the user that his idea will be changed and he could anytime end the game etc...

i put all adjusted things in a new commit to make it easier to review the new changes.
but overall i'm sure it works now a bit smoother.

Comment (anschnapp) [reply agents/devils-advocate.agent.md]: wait something went wrong, i need to correct first

Comment (anschnapp) [reply agents/devils-advocate.agent.md]: ok i accidentially got some duplication fragments in my agent in my last commit

i corrected it. so now only relevant changes are there. so it's ready for re review.
